### DISCUSSION TITLE
Building Dev Containers doesn't ask for template options anymore

### DISCUSSION THREAD

**Discussion by SGr33n** [devcontainers/community#151] [General]
Hi,
Since a few weeks, when I build a new devcontainer (also from the official collection) it doesn't ask for options contained in devcontainer-template.json anymore. Is this just a my problem? E.g. javascript-node-mongo have an imageVariant option, and VSCode doesn't ask me for a choice anymore. This also happens on every container having options.

Thanks!

**Comment by jpawlowski** [comment]
same here.

**Comment by bamurtaugh** [comment]
Thanks for opening. Is this the option selection you're referring to?
![image](https://github.com/user-attachments/assets/0548007b-ac2a-4c7c-a262-46f1e73f6dd8)

If so, I just saw this ^ in my VS Code. Are you still able to reproduce this in the latest versions of VS Code and the Dev Containers extension?

**Comment by SGr33n** [comment]
Hi,
Thanks for your reply, it's still happening, VSCode doesn't ask for options in devcontainer-template.json.
This is my VSCode:

Version: 1.92.2 (user setup)
Commit: fee1edb8d6d72a0ddff41e5f71a671c23ed924b9
Date: 2024-08-14T17:29:30.058Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631

Dev containers version is v0.380.0.

**Reply by bamurtaugh** [reply]
Thanks for checking. cc @chrmarti would love your thoughts here

**Reply by SGr33n** [reply]
Thanks for your interest :)

**Reply by chrmarti** [reply]
Not sure what could go wrong here. It works for me on Windows and macOS with javascript-node-mongo.

Could you go back in the Dev Containers versions and check when it was introduced?

Also cc @joshspicer.

**Reply by SGr33n** [reply]
Thanks @chrmarti, I'm going to test it also on macOS (I don't use Docker there because it's a MacBook Air... in the meanwhile how can I install an older Dev Containers version in order to make this test?

**Reply by joshspicer** [reply]
Thanks for helping us out in root causing this @SGr33n!  Navigating to the extension and clicking the little chevron next to "Uninstall" will let you install previous extension versions.

![image](https://github.com/user-attachments/assets/395af4ac-e645-4155-993d-22a8d7bf2c88)

**Reply by joshspicer** [reply]
Ah, I see now that you've already discovered this further down in the discussion :)

**Reply by SGr33n** [reply]
Sorry, I missed this reply :P

**Comment by chrmarti** [comment]
You can use: `F1` > `Extensions: Install Specific Version of Extension...` > `Dev Containers` > Version. And then `F1` > `Developer: Restart Extension Host` to make the chosen version run.

**Comment by SGr33n** [comment]
Thanks @chrmarti , tried 0.369 and it still doesn't work. In the meanwhile I tried it on my MacBook Air and, even if I have synced options it works good on the Mac... could it be an option to automatically choice the proposed option?
Thanks!

**Comment by chrmarti** [comment]
Please append the Dev Containers log after you run the config UI. (`F1` > `Dev Containers: Show Container Log`)

**Reply by SGr33n** [reply]
> Please append the Dev Containers log after you run the config UI. (`F1` > `Dev Containers: Show Container Log`)

Sure, here it is.
Thanks!
[container-log.txt](https://github.com/user-attachments/files/16850309/container-log.txt)

**Reply by chrmarti** [reply]
That seems to be from when the dev container is built and connected to. Could you try with `F1` > `Dev Containers: Add Dev Container Configuration Files...`?

**Reply by SGr33n** [reply]
> That seems to be from when the dev container is built and connected to. Could you try with `F1` > `Dev Containers: Add Dev Container Configuration Files...`?

You're right, it's working if starting from the Project Folder :) Thanks!

**Reply by chrmarti** [reply]
How did you connect to the dev container when you saw the issue?

**Reply by SGr33n** [reply]
> How did you connect to the dev container when you saw the issue?

I was starting from an empty workspace, then F1 > Dev Containers: New Dev Container
